### PR TITLE
Version 2.0.0

### PR DESCRIPTION
**Pull request by smittytone**
Some minor tweaks, and version and name changes to bring library into line with EI standard naming.

### CODE REVIEW

**Comment by smittytone** [comment]
Reviewers: only small changes here, but a quick check would be good and, once merged to master, uploading to the server.

Thanks!

File: README.md [hunk 1]
```diff
@@ -8,132 +8,192 @@ The Dark Sky API returns a wealth of data (in JSON format). As such, it is left
...
-                local message = "Outlook: " + sendData.cast + ". Temperature: " + format("%.1f", celsius) + "ºC";
-                server.log(message);
-            }
-        }
-
-        if ("callCount" in data) server.log("Current Dark Sky API call tally: " + data.callCount + "/1000");
+  if (err) server.error(err);
+
+  if (data) {
+    server.log("Weather forecast data received from Dark Sky");
```
Comment (ppetrosh) [inline README.md]: https://developer.electricimp.com/squirrel/codestyle:

Indentations should never be tabs (\t) and should always be four spaces.

File: DarkSky.agent.lib.nut [hunk 2]
```diff
@@ -200,17 +222,16 @@ class DarkSky {
                 return a.tointeger();
             }
         }
-
         return -1;
     }
 
-    function _checkCoords(longitude, latitude, caller) {
+    function _checkCoords(longitude = 999.0, latitude = 999.0, caller = "function") {
```
Comment (ppetrosh) [inline DarkSky.agent.lib.nut:313]: Why do we change it to `float`?

File: DarkSky.agent.lib.nut [hunk 3]
```diff
@@ -10,43 +10,49 @@ class DarkSky {
...
+
+        // Set instance properties
+        if (typeof debug != "bool") debug = false;
         _debug = debug;
-        _apikey = key;
         _units = "auto";
+        _apikey = key;
     }
 
     function forecastRequest(longitude = 999, latitude = 999, callback = null) {
```
Comment (ppetrosh) [inline DarkSky.agent.lib.nut]: ... and why don't we change it here?

Comment (smittytone) [reply DarkSky.agent.lib.nut]: These should be floats, so I'll update to be consistent.

File: README.md (hunk 1 above)
Comment (smittytone) [reply README.md]: I thought they were. Old-ish code, so I'll clean out the tabs.

**Comment by smittytone** [comment]
@ppetrosh A reminder for this update too. It's modified as per your earlier comments; be good to get this out.

**Comment by ppetrosh** [comment]
As we are bumping up the major version does it make sense to rename the library to `DarkSky.agent.lib.nut` according to https://developer.electricimp.com/libraries/submissions?

**Comment by smittytone** [comment]
@ppetrosh I've come back to this after some time away. I've dealt with the earlier issues, and found and fixed, a couple of other ones, so it would be good to get this update finished and uploaded, please.

**Pull request closed without merging**